File: 1996-the-number-of-weak-characters-in-the-game/1996-the-number-of-weak-characters-in-the-game.cpp
class Solution {
public:
    int numberOfWeakCharacters(vector<vector<int>>& properties) {
        sort(properties.begin(), properties.end(),
             [&](const vector<int> &a, const vector<int> &b) {
            return a[0] < b[0] || ( a[0] == b[0] && a[1] > b[1] );
        });
        int maxDefence = 0;
        auto isWeak = [&](vector<int>& a) -> bool {
            maxDefence = max(maxDefence, a[1]);
            return a[1] < maxDefence ? true : false;
        };
        int res = 0;
        for(int i = size(properties) - 1; i >= 0; i--) if (isWeak(properties[i])) res++;
        return res;
    }
};

File: 1996-the-number-of-weak-characters-in-the-game/1996-the-number-of-weak-characters-in-the-game.rs
impl Solution {
    pub fn number_of_weak_characters(mut properties: Vec<Vec<i32>>) -> i32 {
        const IDX_ATTACK: usize = 0;
        const IDX_DEFENSE: usize = 1;
        properties.sort_unstable_by_key(|p| (p[IDX_ATTACK], -p[IDX_DEFENSE]));
        let mut maxDef = 0;
        let mut count = 0;
        for idx in (0..properties.len()).rev() {
            let def = properties[idx][IDX_DEFENSE];
            count += (def < maxDef) as i32;
            maxDef = maxDef.max(def);
        }
        count
    }
}